tensorflow/c/kernels/summary_op.cc

```diff
@@ -0,0 +1,177 @@
+      TF_TensorData(params.tags));
+  auto values_array = static_cast<T*>(TF_TensorData(params.values)); 
+  // Copy tags and values into summary protobuf 
+  for (int i = 0; i < TF_TensorElementCount(params.tags); ++i) { 
+    tensorflow::Summary::Value* v = s.add_value();
+    const tensorflow::tstring& Ttags_i = tags_array[i];  
+    v->set_tag(Ttags_i.data(), Ttags_i.size());
+    v->set_simple_value(float(values_array[i]));
+  }
+  TF_Tensor* summary_tensor = TF_AllocateOutput(ctx, 0,
+      TF_ExpectedOutputDataType(ctx, 0), nullptr, 0, 
+      sizeof(tensorflow::tstring), params.status);
+  if (TF_GetCode(params.status) != TF_OK){ 
+    TF_DeleteTensor(summary_tensor); 
+    TF_OpKernelContext_Failure(ctx, params.status);
+    return; 
+  }
+  tensorflow::tstring* output_tstring = reinterpret_cast<tensorflow::tstring*>(
+      TF_TensorData(summary_tensor)); 
+  SerializeToTString(s, output_tstring);
+  TF_DeleteTensor(summary_tensor);
+}
+
+bool IsSameSize(TF_Tensor* tensor1, TF_Tensor* tensor2){ 
+  if (TF_NumDims(tensor1) != TF_NumDims(tensor2)) {
+    return false; 
+  }
+  for (int d = 0; d < TF_NumDims(tensor1); d++) {
+    if (TF_Dim(tensor1, d) != TF_Dim(tensor2, d)) {
+      return false; 
+    }
+  }
+  return true; 
+}
+
+static tensorflow::string SingleTag(TF_Tensor* tags){ 
+  if (TF_TensorElementCount(tags) == 1) { 
+    const char* single_tag = static_cast<tensorflow::tstring*>(
+        TF_TensorData(tags))->c_str(); 
+    return tensorflow::strings::StrCat(" (tag '", single_tag, "')");
+  } 
+  else {
+    return ""; 
+  }
+}
+
+template <typename T>
+void RegisterSummaryScalarOpKernel() {
+  TF_Status* status = TF_NewStatus();
+  {
+    auto* builder = TF_NewKernelBuilder("SummaryScalar", 
+                                        tensorflow::DEVICE_CPU,
+                                        &SummaryScalarOp_Create, 
+                                        &SummaryScalarOp_Compute<T>,
+                                        &SummaryScalarOp_Delete);
+    TF_KernelBuilder_TypeConstraint(builder, "T", 
+        static_cast<TF_DataType>(tensorflow::DataTypeToEnum<T>::v()), status); 
+    CHECK_EQ(TF_OK, TF_GetCode(status))
+        << "Error while adding type constraint";
+    TF_RegisterKernelBuilder("SummaryScalar", builder, status);
+    CHECK_EQ(TF_OK, TF_GetCode(status))
+        << "Error while registering Summary Scalar kernel";
+  }
+  TF_DeleteStatus(status);
+}
+
+// A dummy static variable initialized by a lambda whose side-effect is to
+// register the bitcast kernel.                                                          
+TF_ATTRIBUTE_UNUSED static bool  IsSummaryScalarOpKernelRegistered = []() {                  
+  if (SHOULD_REGISTER_OP_KERNEL("SummaryScalar")) {                                                                           
+    RegisterSummaryScalarOpKernel<tensorflow::int64>();          
+    RegisterSummaryScalarOpKernel<tensorflow::int32>();   
```

Comment:
I see current SummaryScalar also registered for `uint64`, `uint32`, `int8`.
(`TF_CALL_REAL_NUMBER_TYPES` here https://osscs.corp.google.com/tensorflow/tensorflow/+/master:tensorflow/core/kernels/summary_op.cc;l=123?q=%22TF_CALL_REAL_NUMBER_TYPES%22%20summary_op calls `TF_CALL_INTEGRAL` types which includes uint64, uint32, int8)